clang-tools-extra/clang-doc/tool/ClangDocMain.cpp

```diff
@@ -245,10 +247,29 @@ sortUsrToInfo(llvm::StringMap<std::unique_ptr<doc::Info>> &USRToInfo) {
   }
 }
 
+llvm::Error runMappingPhase(tooling::ToolExecutor *Executor,
+                            clang::doc::ClangDocContext &CDCtx,
+                            tooling::ArgumentsAdjuster &ArgAdjuster,
+                            bool IgnoreMappingFailures) {
+  auto Err = Executor->execute(doc::newMapperActionFactory(CDCtx), ArgAdjuster);
+  if (Err) {
+    if (IgnoreMappingFailures) {
+      llvm::errs() << "Error mapping decls in files. Clang-doc will ignore "
+                      "these files and continue:\n"
+                   << toString(std::move(Err)) << "\n";
+      return llvm::Error::success();
+    }
+    return Err;
+  }
+  return llvm::Error::success();
+}
```

Comment:
When I described this before, I thought I had mentioned using a helper for the error handling.

So something along these lines that still makes what's happening clear, but hides the details about IngoreMappingBehavior.

```cpp
ExitOnErr(helper(Executor->execute(doc::newMapperActionFactory(CDCtx), ArgAdjuster));
```